src/Compilers/CSharp/Portable/Binder/InContainerBinder.cs

```diff
@@ -96,6 +96,36 @@ internal override Imports GetImports(ConsList<Symbol> basesBeingResolved)
             return _lazyImports;
         }
 
+        /// <summary>
+        /// Look for a type forwarder for the given type in any referenced assemblies.
+        /// </summary>
+        /// <param name="name">The name of the (potentially) forwarded type.</param>
+        /// <param name="qualifierOpt">Will be used to return the namespace of the found forwarder, 
+        /// if any.</param>
+        /// <param name="diagnostics">Will be used to report non-fatal errors during look up.</param>
+        /// <param name="location">Location to report errors on.</param>
+        /// <returns></returns>
+        /// <remarks>
+        /// Since this method is intended to be used for error reporting, it stops as soon as it finds
+        /// any type forwarder (or an error to report). It does not check other assemblies for consistency or better results.
+        /// </remarks>
+        protected override AssemblySymbol GetForwardedToAssemblyInUsingNamespaces(string name, ref NamespaceOrTypeSymbol qualifierOpt, DiagnosticBag diagnostics, Location location)
+        {
+            var imports = GetImports(basesBeingResolved: null);
+            foreach (var typeOrNamespace in imports.Usings)
+            {
+                var fullName = typeOrNamespace.NamespaceOrType + "." + name;
+                var result = GetForwardedToAssembly(fullName, diagnostics, location);
+                if (result != null || diagnostics.HasAnyErrors())
```

Comment:
>diagnostics.HasAnyErrors() [](start = 38, length = 26)

The behavior of this method should not depend on the contents of the incoming diagnostic bag.